File: SamsidParty.Subsonic.Proxy.AppleMusic/Types/Artist.cs
using Newtonsoft.Json;

namespace SamsidParty.Subsonic.Proxy.AppleMusic.Types
{
    public class Artist : Resource
    {
        [JsonProperty("attributes")]
        public ArtistAttributes Attributes { get; set; }

        [JsonProperty("relationships")]
        public ArtistRelationships Relationships { get; set; }

        [JsonProperty("type")]
        public new string Type { get; set; } = "artists";
    }

    public class ArtistAttributes
    {
        [JsonProperty("editorialNotes")]
        public EditorialNotes EditorialNotes { get; set; }

        [JsonProperty("genreNames")]
        public System.Collections.Generic.List<string> GenreNames { get; set; }

        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("url")]
        public string Url { get; set; }
    }
}

File: FocalSonic/src-vite/src/types/applemusic/artist.ts
import { Resource } from "i18next";
import { ISimilarArtist } from "../responses/artist";
import { AppleMusicAlbum } from "./albums";
import { AppleMusicEditorialNotes, AppleMusicGenre, AppleMusicRelationship } from "./common";

export interface AppleMusicArtist extends Resource {
    attributes?: {
        editorialNotes?: AppleMusicEditorialNotes | undefined;
        genreNames: string[];
        name: string;
        url: string;
    } | undefined;
    relationships?: AppleMusicArtistRelationships | undefined;
    type: "artists";
}

export interface AppleMusicArtistRelationships {
    albums: AppleMusicRelationship<AppleMusicAlbum>;
    genres?: AppleMusicRelationship<AppleMusicGenre> | undefined;
}

export function convertAppleMusicArtistToSubsonic(artist: AppleMusicArtist, appleMusicData: any): ISimilarArtist {
    if (!artist) { return; }
    return {
        id: artist.id,
        name: artist.attributes?.name,
        coverArt: artist.attributes?.artwork?.url || "",
        artistImageUrl: artist.attributes?.artwork?.url || "",
        appleMusic: appleMusicData || { data: [artist] }
    };
}